src/index.ts

```diff
@@ -12,6 +12,72 @@ export interface AccessControlOptions {
   preflightContinue?: boolean
 }
 
+function isIterable(obj: unknown): obj is Iterable<unknown> {
+  return typeof obj[Symbol.iterator] === 'function'
+}
+
+function getOriginHeaderHandler(origin: unknown): (req: Request, res: Response) => void {
+  function fail() {
+    throw new TypeError('No other objects allowed. Allowed types is array of strings or RegExp')
+  }
+
+  if (typeof origin === 'boolean' && origin === true) {
+    return function (_, res) {
+      res.setHeader('Access-Control-Allow-Origin', '*')
+    }
+  }
+
+  if (typeof origin === 'boolean' && origin === false) {
+    return () => undefined
+  }
+
+  if (typeof origin === 'string') {
+    return function (_, res) {
+      res.setHeader('Access-Control-Allow-Origin', origin)
+    }
+  }
+
+  if (typeof origin === 'function') {
+    return function (req, res) {
```

Comment:
use arrow functions pls

Reply:
I used the `function` syntax because I think it is more readable. 
Again, I'll make the change if you're sure, but I really think using arrow functions here reduces the quality of the code